app/models/task.rb

```diff
@@ -14,16 +18,32 @@ class Task < ApplicationRecord
   scope :priority, -> { order(priority: :desc) }
   scope :recent, -> { order(created_at: :desc) }
 
-  def self.search(current_user,input_name,input_status,page_number)
-      if input_name.present? && input_status.present?
-        current_user.tasks.page(page_number).name_like(input_name).where(status:input_status)
-      elsif input_name.blank? && input_status.present?
-        current_user.tasks.page(page_number).where(status:input_status)
-      elsif input_name.present? && input_status.blank?
-        current_user.tasks.page(page_number).name_like(input_name)
-      else
-        current_user.tasks.all.page(page_number).recent
-      end
+  def self.filter(current_user,input_name,input_status,input_label,page_number)
+
+    if input_label.present?
+      tasks = current_user.labels.find_by(name:input_label).tasks
+      search(tasks,input_name,input_status,page_number)
+    else
+      tasks = current_user.tasks
+      search(tasks,input_name,input_status,page_number)
+    end
+
```

Comment:
search部分が同じ処理の場合、下記の方が意図が伝わりやすいです。三項演算子を使っても良いかもしれません。


```suggestion
tasks = if xxx
         else
         end
search(tasks, xxx)
```

Reply:
0d21098 修正